docs/guide/warnings.rst

```diff
@@ -727,6 +727,17 @@ List Of Warnings
    implicit definition of wire '...'".
 
 
+.. option:: IMPLICITSTATIC
+
+   Warns that a lifetime of a task or a function is implicitly set to
+   static. It may be confusing for C users and lead to bugs. The warning
+   isn't thrown when no variable inside a task or a function has value
+   assignment in its declaration.
+
+   Ignoring this warning will only suppress the lint check; it will
+   simulate correctly.
```

Comment:
```suggestion
   Warns that the lifetime of a task or a function was not provided and so
   was implicitly set to static. The warning is suppressed when no
   variables inside the task or a function are assigned to.

   This is a warning because the static default differs from C++, differs
   from class member function/tasks.  Static is a more dangerous default
   then automatic as static prevents the function from being reinterant,
   which may be a source of bugs, and/or performance issues.

   If the function does not require static behavior, change it to "function
   automatic".

   If the function requires static behavior, change it to "function
   static".

   Ignoring this warning will only suppress the lint check; it will
   simulate correctly.
```

Also update docs for warnings that --Wlint does.